src/adaptix/_internal/morphing/generic_provider.py

```diff
@@ -99,19 +100,95 @@ def _get_allowed_values_collection(self, args: Collection) -> Collection:
             return set(args)
         return tuple(args)
 
+    def _get_allowed_values_repr(self, args: Collection, mediator: Mediator, loc_stack: LocStack) -> Collection:
+        enum_cases = [arg for arg in args if isinstance(arg, Enum)]
+        if not enum_cases:
+            return set(args)
+
+        literal_dumper = self._provide_dumper(mediator, DumperRequest(loc_stack))
+        return {literal_dumper(arg) if isinstance(arg, Enum) else arg for arg in args}
+
+    def _get_enum_types(self, cases: Collection) -> Collection:
+        seen: Set[Type[Enum]] = set()
+        seen_add = seen.add
+        return [type(case) for case in cases if not (type(case) in seen or seen_add(case))]
```

Comment:
This is a very strange that mutating method is used inside list comprehension. It looks like for-loop will be more obvious